pycqed/scripts/Spectroscopy.py

```diff
@@ -0,0 +1,281 @@
+        self.freq_range = self.options_dict.get('freq_range',None)
+        self.amp_range = self.options_dict.get('amp_range',None)
+        self.phase_range = self.options_dict.get('phase_range',None)
+        self.plotsize = self.options_dict.get('plotsize',(8,5))
+        if len(self.data_dict['timestamps']) == 1:
+            self.plot_frequency = np.squeeze(self.data_dict['freq'])
+            self.plot_amp = np.squeeze(self.data_dict['amp'])
+            self.plot_phase = np.squeeze(self.data_dict['phase'])
+            print(self.plot_frequency.shape, self.plot_amp.shape, self.plot_phase.shape)
+
+        else:
+            #TRANSPOSE ALSO NEEDS TO BE CODED FOR 2D
+            self.plot_xvals = self.options_dict.get('xvals',np.array([[tt] for tt in range(len(self.data_dict['timestamps']))]))
+            self.plot_xwidth = self.options_dict.get('xwidth',None)
+            self.plot_xlabel = self.options_dict.get('xlabel','Scan number')
+            if self.plot_xwidth == 'auto':
+                x_diff = np.diff(np.ravel(self.plot_xvals))
+                dx1 = np.concatenate(([x_diff[0]],x_diff))
+                dx2 = np.concatenate((x_diff,[x_diff[-1]]))
+                self.plot_xwidth = np.maximum(dx1,dx2)
+                self.plot_frequency = np.array([[tt] for tt in self.data_dict['freq']])
+                self.plot_phase = np.array([[tt] for tt in self.data_dict['phase']])
+                self.plot_amp = np.array([np.array([tt]).transpose() for tt in self.data_dict['amp']])
+
+            else:
+                self.plot_frequency = self.data_dict['freq']
+                self.plot_phase = self.data_dict['phase']
+                self.plot_amp = self.data_dict['amp']
+
+
+
+
+
+
+
+    def prepare_plots(self):
+
+        if len(self.data_dict['timestamps']) == 1:
+            plot_fn = self.plot_line
+            self.plot_dicts['amp'] = {'plotfn': plot_fn,
+                            'xvals': self.plot_frequency,
+                            'yvals': self.plot_amp,
+                            'title': 'Spectroscopy amplitude: %s'%(self.timestamps[0]),
+                            'xlabel': self.plot_xlabel,
+                            'ylabel': self.amp_label,
+                            'yrange': self.amp_range,
+                            'plotsize': self.plotsize
+                            }
+            self.plot_dicts['phase'] = {'plotfn': plot_fn,
+                            'xvals': self.plot_frequency,
+                            'yvals': self.plot_phase,
+                            'title': 'Spectroscopy phase: %s'%(self.timestamps[0]),
+                            'xlabel': self.plot_xlabel,
+                            'ylabel': self.phase_label,
+                            'yrange': self.phase_range,
+                            'plotsize': self.plotsize
+                            }
+        else:
+            print('plotting not yet coded up for multiple traces')
```

Comment:
replace `print` with `logging.warning` (may need `import logging` at the top of the file)